pkg-manager/plugin-commands-installation/test/catalog.ts

```diff
@@ -0,0 +1,324 @@
+    packages: ['packages/*'],
+    catalog: {
+      'is-positive': '^1.0.0',
+      'is-negative': '^1.0.0',
+    },
+  })
+
+  await ctrl.install()
+  const lockfile = await ctrl.lockfile()
+
+  expect(lockfile.catalogs).toStrictEqual({
+    default: {
+      'is-positive': { specifier: '^1.0.0', version: '1.0.0' },
+      'is-negative': { specifier: '^1.0.0', version: '1.0.0' },
+    },
+  })
+})
+
+test('lockfile is updated if catalog config changes', async () => {
+  const ctrl = new CatalogTestsController([
+    {
+      location: 'packages/project1',
+      package: {
+        dependencies: {
+          'is-positive': 'catalog:',
+        },
+      },
+    },
+  ])
+
+  await ctrl.writeWorkspaceYml({
+    packages: ['packages/*'],
+    catalog: { 'is-positive': '=1.0.0' },
+  })
+  await ctrl.install()
+
+  expect((await ctrl.lockfile()).importers['packages/project1']).toEqual({
+    dependencies: {
+      'is-positive': {
+        specifier: 'catalog:',
+        version: '1.0.0',
+      },
+    },
+  })
+
+  await ctrl.writeWorkspaceYml({
+    packages: ['packages/*'],
+    catalog: {
+      'is-positive': '=3.1.0',
+    },
+  })
+  await ctrl.install()
+
+  expect((await ctrl.lockfile()).importers['packages/project1']).toEqual({
+    dependencies: {
+      'is-positive': {
+        specifier: 'catalog:',
+        version: '3.1.0',
+      },
+    },
+  })
+})
+
+test('lockfile catalog snapshots retain existing entries on --filter', async () => {
+  const ctrl = new CatalogTestsController([
+    {
+      location: 'packages/project1',
+      package: {
+        dependencies: {
+          'is-negative': 'catalog:',
+        },
+      },
+    },
+    {
+      location: 'packages/project2',
+      package: {
+        dependencies: {
+          'is-positive': 'catalog:',
+        },
+      },
+    },
+  ])
+
+  await ctrl.writeWorkspaceYml({
+    packages: ['packages/*'],
+    catalog: {
+      'is-positive': '^1.0.0',
+      'is-negative': '^1.0.0',
+    },
+  })
+
+  await ctrl.install()
+
+  expect((await ctrl.lockfile()).catalogs).toStrictEqual({
+    default: {
+      'is-negative': { specifier: '^1.0.0', version: '1.0.0' },
+      'is-positive': { specifier: '^1.0.0', version: '1.0.0' },
+    },
+  })
+
+  // Update catalog definitions so pnpm triggers a rerun.
+  await ctrl.writeWorkspaceYml({
+    packages: ['packages/*'],
+    catalog: {
+      'is-positive': '=3.1.0',
+      'is-negative': '^1.0.0',
+    },
+  })
+  await ctrl.install({ filter: ['packages/project2'] })
+
+  expect((await ctrl.lockfile()).catalogs).toStrictEqual({
+    default: {
+      // The is-negative snapshot should be carried from the previous install,
+      // despite the current filtered install not using it.
+      'is-negative': { specifier: '^1.0.0', version: '1.0.0' },
+
+      'is-positive': { specifier: '=3.1.0', version: '3.1.0' },
+    },
+  })
+})
+
+test('lockfile catalog snapshots should keep unused entries', async () => {
```

Comment:
Intentionally keeping around unused entries for now. Otherwise we'd wipe out `catalog:` dependencies in a filtered install. I think we'll want to change this logic in another PR to prevent stale references though.